migration/migration.go

```diff
@@ -576,43 +580,92 @@ func createOrUpdateWorkItemLinkType(ctx context.Context, linkCatRepo *link.GormW
+// createCategory creates category in db
+func createOrUpdateSingleCategory(ctx context.Context, categoryRepo category.Repository, categoryID *uuid.UUID, categoryName string) error {
+	category := category.Category{
+		ID:   *categoryID,
+		Name: categoryName,
+	}
+	_, err := categoryRepo.Create(ctx, &category)
+	if err != nil {
+		return errs.WithStack(err)
+	}
+	return nil
+}
+
+func createOrUpdateCategories(ctx context.Context, db *gorm.DB, categoryRepo category.Repository, categoryID *uuid.UUID, categoryName string) error {
+	cat, err := categoryRepo.LoadCategoryFromDB(ctx, *categoryID)
+	cause := errs.Cause(err)
+	switch cause.(type) {
+	case errors.NotFoundError:
+		err = createOrUpdateSingleCategory(ctx, categoryRepo, categoryID, categoryName)
+		if err != nil {
+			return errs.WithStack(err)
+		}
+	case nil:
+		log.Info(ctx, map[string]interface{}{
+			"category": categoryName,
+		}, "Category %s exists, will update/overwrite all fields", categoryName)
+		cat.ID = *categoryID
+		cat.Name = categoryName
+		db = db.Save(cat)
+		return errs.WithStack(err)
+	}
+	return nil
+}
+
+// populateCategories populates the categories table with system defined categories
+func populateCategories(ctx context.Context, db *gorm.DB, categoryRepo category.Repository) {
+	fmt.Println("Creating or updating categories...")
+	createOrUpdateCategories(ctx, db, categoryRepo, &category.PlannerRequirementsID, category.PlannerRequirements)
+	createOrUpdateCategories(ctx, db, categoryRepo, &category.PlannerIssuesID, category.PlannerIssues)
+	fmt.Println("Creating/updating of categories done.")
+}
+
 // PopulateCommonTypes makes sure the database is populated with the correct types (e.g. bug etc.)
 func PopulateCommonTypes(ctx context.Context, db *gorm.DB, witr *workitem.GormWorkItemTypeRepository) error {
 	populateLocker.Lock()
 	defer populateLocker.Unlock()
 	if err := createSpace(ctx, space.NewRepository(db), space.SystemSpace, "The system space is reserved for spaces that can to be manipulated by the user."); err != nil {
 		return errs.WithStack(err)
 	}
-	if err := createOrUpdateSystemPlannerItemType(ctx, witr, db, space.SystemSpace); err != nil {
+
+	c := category.NewRepository(db)
+	populateCategories(ctx, db, c) // populate the categories
+	categories, err := c.List(ctx) // fetch the categories
+	if err != nil {
+		return errs.WithStack(err)
+	}
+	if err := createOrUpdateSystemPlannerItemType(ctx, witr, db, space.SystemSpace, categories[0].ID); err != nil {
 		return errs.WithStack(err)
 	}
 	workitem.ClearGlobalWorkItemTypeCache() // Clear the WIT cache after updating existing WITs
-	if err := createOrUpdatePlannerItemExtension(workitem.SystemUserStory, "User Story", "Desciption for User Story", "fa-map-marker", ctx, witr, db, space.SystemSpace); err != nil {
+	if err := createOrUpdatePlannerItemExtension(workitem.SystemUserStory, "User Story", "Desciption for User Story", "fa-map-marker", ctx, witr, db, space.SystemSpace, categories[0].ID); err != nil {
```

Comment:
Why use `categories[0].ID` instead of the IDs (e.g. `PlannerRequirementsID`) that you hard coded into the repository?